sdk/search/Azure.Search.Documents/src/Indexes/SearchIndexerClient.cs

```diff
@@ -248,9 +249,37 @@ public class SearchIndexerClient
         /// </returns>
         /// <exception cref="ArgumentNullException">Thrown when <paramref name="dataSourceConnection"/> is null.</exception>
         /// <exception cref="RequestFailedException">Thrown when a failure is returned by the Search service.</exception>
+        [EditorBrowsable(EditorBrowsableState.Never)]
+#pragma warning disable AZC0002 // DO ensure all service methods, both asynchronous and synchronous, take an optional CancellationToken parameter called cancellationToken.
+        public virtual Response<SearchIndexerDataSourceConnection> CreateOrUpdateDataSourceConnection(
+#pragma warning restore AZC0002 // DO ensure all service methods, both asynchronous and synchronous, take an optional CancellationToken parameter called cancellationToken.
+            SearchIndexerDataSourceConnection dataSourceConnection,
+            bool onlyIfUnchanged,
+            CancellationToken cancellationToken)
+        {
+            return CreateOrUpdateDataSourceConnection(dataSourceConnection, onlyIfUnchanged, null, cancellationToken);
```

Comment:
nit - you might save yourself some pain by doing
```suggestion
            return CreateOrUpdateDataSourceConnection(dataSourceConnection, onlyIfUnchanged, ignoreResetRequirements: null, cancellationToken);
```

so this stays unambiguous if you ever add another optional param in the future.

Reply:
Made the change for all overloads.